docs/architecture/event-flows-proposal.md

```diff
@@ -0,0 +1,85 @@
+# Eventing flow proposal
+
+## Overview
+
+This document shows a proposal of two ways of eventing flow we want to support with use of the [Management Plane](./terminology.md#management-plane)
+
+## Eventing with Kyma internal Event Bus
+
+![Management Plane Components](assets/internal-event-flow.svg)
+
+1. The event is registered using **Management Plane** API.
+2. The **Director** sends an event url to Application.
+3. The **Agent** fetches the event data and passes it to Kyma Event Bus through Event Gateway. It will support multiple protocols like HTTP, MQTT along with multiple authentication (e.g. OpenID Connect) and authorization (e.g. OAuth2) methods.  
+4. The Application publishes an event.
+
+## Eventing with external Event Bus
+
+![Management Plane Components](assets/external-event-flow.svg)
+
+1. The event is registered using **Management Plane** API with external event bus data.
+2. The **Agent** fetches the event and passes information to eventing system (E.g Google's Cloud PubSub).
+3. Eventing system subscribes on external event bus.
+4. The application publishes an event.
+
+## GraphQL Schema  
+
+### Types
+```graphql
+
+type EventAPIDefinition {
+    id: ID!
+    """group allows you to find the same API but in different version"""
+    group: String
+    spec: EventSpec!
```

Comment:
what is "EventSpecType" in details?

Reply:
It was already in the schema, but I will add it to proposal too